privaterelay/settings.py

```diff
@@ -95,9 +96,14 @@
     r"^__heartbeat__",
     r"^__lbheartbeat__",
 ]
+SECURE_HSTS_INCLUDE_SUBDOMAINS = config(
+    "DJANGO_SECURE_HSTS_INCLUDE_SUBDOMAINS", False, cast=bool
```

Comment:
question (blocking): We're setting this to `False` here, but are we planning to set it to `True` in production? The warning in the docs ("Setting this incorrectly can irreversibly (for the value of [SECURE_HSTS_SECONDS](https://docs.djangoproject.com/en/4.2/ref/settings/#std-setting-SECURE_HSTS_SECONDS)) break your site.") makes me nervous about setting this on any *.firefox.com subdomain. If a client sees this parameter on relay.firefox.com, could a client apply this to accounts.firefox.com, data.firefox.com, etc.?

Reply:
No, this would only affect relay.firefox.com and subdomains under relay.firefox.com, such as www.relay.firefox.com. The specification at [RFC 6797 Section 8.2 Known HSTS Host Domain Name Matching](https://www.rfc-editor.org/rfc/rfc6797#section-8.2) has some examples. The domains accounts.firefox.com or data.firefox.com are not affected, because they are not subdomains of relay.firefox.com.

Our current header on relay.firefox.com is:

```
strict-transport-security: max-age=31536000
```

This tells browsers to remember to only use HTTPS on relay.firefox.com for 1 year. https://hstspreload.org/ recommends 2 years. We can lower this age while we are testing changes, such as reducing it to 300 (5 minutes)

The current header for firefox.com is:

```
strict-transport-security: max-age=0
```

According to [RFC 6797 Section 6.2 Examples](www.rfc-editor.org/rfc/rfc6797#section-6.2), this is interpreted as:

> The HSTS header field ... indicates that the UA must delete the entire HSTS Policy associated with the HSTS Host that sent the header field

This does not include `includeSubDomains`, so it says nothing about relay.firefox.com, data.firefox.com, etc.

accounts.firefox.com uses 1 year and includes subdomains:

```
strict-transport-security: max-age=31536000; includeSubDomains
```

data.firefox.com does not set the header.